### PR TITLE
#14908 comment field added to roles for PostgreSQL and some forks

### LINKED ISSUE #14908: PostgreSQL Roles missing comments feature in GUI
When viewing postgresql roles in dbeaver, there is no place where comments are displayed.

**Describe the solution you'd like**
Similar functionality to other objects: Display/Edit/Add comments. Add comments to Generate DDL for Roles

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2022-09-26 16_39_56-test_as - Persist Changes

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java [hunk 1]
```diff
@@ -148,4 +152,21 @@ private void addRoleOptions(StringBuilder script, PostgreRole role, NestedObject
...
     }
+
+    @Override
+    protected void addObjectExtraActions(
+        DBRProgressMonitor monitor,
+        DBCExecutionContext executionContext,
+        List<DBEPersistAction> actions,
+        NestedObjectCommand<PostgreRole, PropertyHandler> command,
+        Map<String, Object> options)
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java:163]:
```suggestion
        Map<String, Object> options
    ) {
```

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreRole.java [hunk 2]
```diff
@@ -631,4 +652,11 @@ public boolean isValidValue(PostgreRole object, Object value) throws IllegalArgu
             return !object.isPersisted();
         }
     }
+
+    public static class DescriptionValidator implements IPropertyValueValidator<PostgreRole, Object> {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreRole.java]:
```suggestion
    public static class CommentsOnRolesSupportedValidator implements IPropertyValueValidator<PostgreRole, Object> {
```

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java [hunk 3]
```diff
@@ -158,6 +158,11 @@ public interface PostgreServerExtension {
      * */
     boolean supportsRoleBypassRLS();
 
+    /**
+     * @return true if database supports syntax like "COMMENT ON ROLE roleName IS 'comment'"
+     */
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java]:
```suggestion
    /**
     * Determines whether the database supports syntax like {@code COMMENT ON ROLE roleName IS 'comment'} or not
     */
```

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java (hunk 3 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java]: Looks like something just for the CheckStyle :(

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java (hunk 1 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java:163]: And now we here again.

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java (hunk 3 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreServerExtension.java]: It is. In fact, having the summary is better that having a dangling `@return` tag IMO 👀

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java:163]: Make CheckStyle happy.
Make me happy.

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java:163]: Nope. Just nope. We discussed it many times.

Map<String, Object> options) <-- end of the arguments list
{ <-- start of the method body

A matter of taste.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreRoleManager.java:163]: I thought we agreed to use the proposed pattern last time :thinking_face:
It's sad

**Review by ShadelessFox — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**